### ISSUE TITLE
add bind-propagation to docker-compose

### ISSUE THREAD

**Issue by rdxmb** [docker/compose#5212]
```
version: '3.3'

services:

  container:
    image: phusion/baseimage
    container_name: mycontainer
    volumes:
    - type: bind
      source: /mnt
      target: /mnt
      bind-propagation: shared
```
the last line seems to be ignored, as the propagation when starting the container is `rprivate` - which is the default. When running the service with

`docker service create --mount type=bind,src=/mnt,target=/mnt,bind-propagation=shared --name bind-propagation phusion/baseimage`

the propagation indeed is `shared`.

```
root@docker-single-host:~# docker-compose --version
docker-compose version 1.16.1, build 6d1ac21
root@docker-single-host:~# docker --version
Docker version 17.06.2-ce, build cec0b72
```

This can be checked via
`docker inspect --format "{{json .Mounts}}" <CONTAINER>`

**Comment by shin-** [comment]
Please report `docker stack` issues on the [moby/moby](https://github.com/moby/moby) repo!

**Comment by rdxmb** [comment]
@shin- just to get this correctly:
I get the unexpected result when running `docker-compose up`. So is this also the wrong project? What type of issues are this project for? (just to make this right the next time...) Thanks!

**Comment by rdxmb** [comment]
https://github.com/moby/moby/issues/34936

**Comment by shin-** [comment]
Yes. Anything that involves adding fields to the v3 format should go through moby/moby or docker/cli first, as the engine is the source of truth for the v3 format as a whole. When changes happen there, we then port them to `docker-compose`.

Thank you!

**Comment by rdxmb** [comment]
As reported with https://github.com/moby/moby/issues/34936, the following syntax is supported with `docker stack`:
```
volumes:
    - type: bind
      bind:
          propagation: shared
```
This is not supported with `docker-compose`.
Inspecting the container with `docker inspect` running with `docker stack`:
```
    ...
            "Mounts": [
                {
                    "Type": "bind",
                    "Source": "/mnt",
                    "Target": "/mnt",
                    "BindOptions": {
                        "Propagation": "shared"
                    }
                }
            ]
        },
        "GraphDriver": {
            "Data": null,
            "Name": "aufs"
        },
        "Mounts": [
            {
                "Type": "bind",
                "Source": "/mnt",
                "Destination": "/mnt",
                "Mode": "",
                "RW": true,
                "Propagation": "shared"
            }
        ],
   ...
```
Inspecting the container with `docker inspect` running with `docker-compose`:
        "Mounts": [
            {
                "Type": "bind",
                "Source": "/mnt",
                "Destination": "/mnt",
                "Mode": "rw",
                "RW": true,
                "Propagation": "rprivate"
            }
        ],

Just tested with
```
root@docker-single-host:~/volumes-test# docker --version
Docker version 17.09.0-ce, build afdb6d4
root@docker-single-host:~/volumes-test# docker-compose --version
docker-compose version 1.16.1, build 6d1ac21
```

and compose-file version 3.3

So could we please reopen this issue?

**Comment by shin-** [comment]
I believe this was fixed in 1.18.0!

**Comment by rdxmb** [comment]
works with docker-compose 1.21.0 .
